### PR TITLE
Add ammo types

### LINKED ISSUE #31: Add ammo types
- [x] Normal
- [x] Armor piercing
- [ ] Explosive

### PR DESCRIPTION
**Pull request by kdhageman**
Fixes #31 

### Normal ammo
Regular bullets, nothing fancy
projectbeard_normal

### Armor piercing ammo
Similar to normal ammo, but bypasses armor
projectbeard_piercing

### Explosive ammo
Creates an explosion when hitting something. The explosion deals damage to ANY tank it hits, including the tank that fired the bullet.
Create explosion on hit
Scale explosion size with tank stats.
projectbeard_explosion

### Projectiles collision
Projectiles now collide with the edge of the terrain and with the obstacles.
projectbeard_collision

## Damage system
```
                                                      +------------------+
                                                      |      Normal      |
                                                      |                  |
                                                 +--> |   = mult * 1     |
                                                 |    |                  |
                                                 |    +------------------+
                                                 |
+-------------+       +----------------------+   |    +------------------+
|             |       |                      |   |    |  Armor Piercing  |
|  TankStats  |       | ProjectileController |   |    |                  |
|  (Damage)   +-----> |    (Multiplier)      +------> |   = mult * 0.9   |
|             |       |                      |   |    |                  |
+-------------+       +----------------------+   |    +------------------+
                                                 |
                                                 |    +------------------+
                                                 |    |    Explosive     |
                                                 |    |                  |
                                                 +--> |   = mult * 0.75  |
                                                      |                  |
                                                      +------------------+

```

The `TankStats` has already defined a projectile damage value. This value is used by the `ProjectileController` as a base damage value. `ProjectileController` adds a `DamageMultiplier` value, which is used in combination with the base damage to compute the actual damage 
```
base damage * damage multiplier = actual damage
```

With the image above in mind, an example; a tank with base damage of `10` and explosive ammo, will hit the enemy for `10 * 0.75 = 7.5` damage.

### CODE REVIEW

**Comment by ratenbuuren** [comment]
Had a quick play-through just now. 
Works as intended, one nice addition:
- Add UI for ammo type?

File: Assets/Scripts/Projectiles/ProjectileController.cs [hunk 1]
```diff
@@ -1,40 +1,44 @@
...
-using UnityEngine;
+﻿using UnityEngine;
 
-public class ProjectileController : MonoBehaviour {
-    private AmmoType ammoType;
-    private float damage = 25f;
-    private float velocity = 3f;
-    private float range = 4f;
+public abstract class ProjectileController : MonoBehaviour {
+    public float damageModifier = 1f;
```
Comment (ratenbuuren) [inline Assets/Scripts/Projectiles/ProjectileController.cs]: No consistent naming (Rider style) in this class

File: Assets/Scripts/Prefabs/DestroyAfterTime.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+using UnityEngine;
+using System.Collections;
+
+public class DestroyAfterTime : MonoBehaviour {
+
+    public float lifetime = 1f;
+
+    // Use this for initialization
```
Comment (ratenbuuren) [inline Assets/Scripts/Prefabs/DestroyAfterTime.cs]: Remove comment

File: Assets/Scripts/GameManager.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
Comment (ratenbuuren) [inline Assets/Scripts/GameManager.cs]: Much better ;)

**Review by ratenbuuren — CHANGES_REQUESTED** [review]
Looks great. Nice setup. Some really minor things for improving next to my other comment.

File: Assets/Scripts/GameManager.cs (hunk 3 above)
Comment (kdhageman) [reply Assets/Scripts/GameManager.cs]: Hm, when I revert this change, Rider does not recognise `NullPointerException`.

**Comment by kdhageman** [comment]
@ratenbuuren Good idea to add ammo type to the UI. It's outside of the scope of this PR, so I'll add a new issue.

**Review by ratenbuuren — APPROVED** [review]

**Pull request merged**